planner/cascades/implementation_rules.go

```diff
@@ -221,3 +224,32 @@ func (r *ImplSort) OnImplement(expr *memo.GroupExpr, reqProp *property.PhysicalP
 	)
 	return impl.NewSortImpl(ps), nil
 }
+
+// ImplHashAgg is the implementation rule which implements LogicalAggregation
+// to PhysicalHashAgg.
+type ImplHashAgg struct {
+}
+
+// Match implements ImplementationRule Match interface.
+func (r *ImplHashAgg) Match(expr *memo.GroupExpr, prop *property.PhysicalProperty) (matched bool) {
+	if !prop.IsEmpty() {
+		return false
+	}
+	la := expr.ExprNode.(*plannercore.LogicalAggregation)
+	_, preferStream := la.ResetHintIfConflicted()
+	if preferStream {
+		// If the LogicalAggregation has a StreamAgg hint but it cannot be implemented as PhysicalStreamAgg,
+		// we will generate a PhysicalHashAgg in the rule `ImplStreamAgg`. So we have no need to regenerate
+		// a HashAgg in this rule.
+		return false
```

Comment:
I'm considering introduce a new part in the implementation phase to deal with hints. 
1. Firstly we can use hint to choose the implementation rule set.
2. If the hint is not applicable, reset the hint and try other implementation rules.
3. If the hint is applicable, we can just skip other implementation rules.

Reply:
Agree, `Match` is not supposed to handle hints.

Reply:
Shall we do this after implementing StreamAgg? It's hard to add unit tests for hints in this PR.